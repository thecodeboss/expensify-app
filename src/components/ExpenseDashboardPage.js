import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseDashboardPage = ({expenses = [], filters}) => (
  <div>
    <ExpenseListFilters />
    {
      expenses.length > 0 &&
      <p>
        Viewing {expenses.length} {expenses.length > 1 ? "expenses " : "expense "}
        totalling {numeral(selectExpensesTotal(expenses) / 100).format('$0,0.00')}.
      </p>
    }
    <ExpenseList />
  </div>
);

const mapStateToProps = ({expenses, filters}) => ({
  expenses: selectExpenses(expenses, filters)
});

export default connect(mapStateToProps)(ExpenseDashboardPage);
