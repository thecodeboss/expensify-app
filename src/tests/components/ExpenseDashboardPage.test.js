import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseDashboardPage } from '../../components/ExpenseDashboardPage.js';
import expenses from '../fixtures/expenses';

test('should render ExpenseDashboardPage correctly', () => {
  const wrapper = shallow(<ExpenseDashboardPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseDashboardPage correctly with 1 expense', () => {
  const wrapper = shallow(<ExpenseDashboardPage expenses={[expenses[0]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseDashboardPage correctly with more than 1 expense', () => {
  const wrapper = shallow(<ExpenseDashboardPage expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});
