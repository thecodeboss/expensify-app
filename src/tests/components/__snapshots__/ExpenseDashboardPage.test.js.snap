// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseDashboardPage correctly 1`] = `
<div>
  <Connect(ExpenseListFilters) />
  <Connect(ExpenseList) />
</div>
`;

exports[`should render ExpenseDashboardPage correctly with 1 expense 1`] = `
<div>
  <Connect(ExpenseListFilters) />
  <p>
    Viewing 
    1
     
    expense 
    totalling 
    $1.95
    .
  </p>
  <Connect(ExpenseList) />
</div>
`;

exports[`should render ExpenseDashboardPage correctly with more than 1 expense 1`] = `
<div>
  <Connect(ExpenseListFilters) />
  <p>
    Viewing 
    3
     
    expenses 
    totalling 
    $1,141.95
    .
  </p>
  <Connect(ExpenseList) />
</div>
`;
